flocker/node/agents/cinder.py

```diff
@@ -852,17 +854,53 @@ def get_keystone_session(**config):
         )
 
 
-def get_cinder_v1_client(session, region):
+SUPPORTED_VERSIONS = (
+    (u"2", u"volumev2"),
+    (u"1", u"volume"),
+)
+
+
+class CinderAPIVersionDetectionFailure(Exception):
```

Comment:
Probably should derive from EnvironmentError.

https://docs.python.org/2/library/exceptions.html#exceptions.EnvironmentError